frontend/src/domain/Role.ts

```diff
@@ -1,4 +1,7 @@
+import type { Knowledge } from './Knowledge'
+
 export interface Role {
   id?: number
   name: string
+  knowledgesOfaRole: Array<Knowledge>
```

Comment:
Since we are inside a role, we already know that these knowledges are property of this role. So, knowledges would be enough
```suggestion
  knowledges: Array<Knowledge>
```